waymo2argo/waymo_raw_data_to_argoverse.py

```diff
@@ -0,0 +1,441 @@
+#!/usr/bin/env python3
+
+import argparse
+import glob
+import imageio
+import itertools
+import json
+import math
+import numpy as np
+import os
+import pandas as pd
+from pathlib import Path
+import pdb
+from typing import Any, Dict, List, Union
+import uuid
+
+import cv2
+from pyntcloud import PyntCloud
+from scipy.spatial.transform import Rotation
+import tensorflow.compat.v1 as tf
+
+tf.enable_eager_execution()
+
+from argoverse.utils.json_utils import save_json_dict
+from argoverse.utils.se3 import SE3
+import waymo_open_dataset
+from waymo_open_dataset.utils import range_image_utils
+from waymo_open_dataset.utils import transform_utils
+from waymo_open_dataset.utils import frame_utils
+from waymo_open_dataset import dataset_pb2 as open_dataset
+
+from waymo2argo.transform_utils import (
+    rotX,
+    rotY,
+    rotmat2quat,
+    quat2rotmat,
+    yaw_to_quaternion3d,
+)
+
+"""
+Extract poses, images, and camera calibration from raw Waymo Open Dataset TFRecords.
+
+See the Frame structure here:
+https://github.com/waymo-research/waymo-open-dataset/blob/master/waymo_open_dataset/dataset.proto
+
+See paper:
+https://arxiv.org/pdf/1912.04838.pdf
+"""
+
+# Mapping from Argo Camera names to Waymo Camera names
+# The indices correspond to Waymo's cameras
+CAMERA_NAMES = [
+    "unknown",  # 0, 'UNKNOWN',
+    "ring_front_center",  # 1, 'FRONT'
+    "ring_front_left",  # 2, 'FRONT_LEFT',
+    "ring_front_right",  # 3, 'FRONT_RIGHT',
+    "ring_side_left",  # 4, 'SIDE_LEFT',
+    "ring_side_right",  # 5, 'SIDE_RIGHT'
+]
+
+# Mapping from Argo Label types to Waymo Label types
+# The indices correspond to Waymo's label types
+LABEL_TYPES = [
+    "OTHER_MOVER",  # 0, TYPE_UNKNOWN
+    "VEHICLE",  # 1, TYPE_VEHICLE
+    "PEDESTRIAN",  # 2, TYPE_PEDESTRIAN
+    "SIGN",  # 3, TYPE_SIGN
+    "BICYCLIST",  # 4, TYPE_CYCLIST
+]
+
+RING_IMAGE_SIZES = {
+    # width x height
+    "ring_front_center": (1920, 1280),
+    "ring_front_left": (1920, 1280),
+    "ring_side_left": (1920, 886),
+    "ring_side_right": (1920, 886),
+}
+
+
+def str2bool(v):
+    if isinstance(v, bool):
+        return v
+    elif v.lower() in ("yes", "true", "t", "y", 1):
+        return True
+    elif v.lower() in ("no", "false", "f", "n", 0):
+        return False
+    else:
+        raise argparse.ArgumentTypeError("Boolean value expected.")
+
+
+def round_to_micros(t_nanos: int, base: int = 1000) -> int:
+    """
+    Round nanosecond timestamp to nearest microsecond timestamp
+    """
+    return base * round(t_nanos / base)
+
+
+def check_mkdir(dirpath: str) -> None:
+    """ """
+    if not Path(dirpath).exists():
+        os.makedirs(dirpath, exist_ok=True)
+
+
+def get_log_id_from_files(record_dir: str) -> List[str]:
+    """Get the log IDs of the Waymo records from the directory
+       where they are stored
+
+    Args:
+        record_dir: The path to the directory where the Waymo data
+                    is stored
+                    Example: "/path-to-waymo-data"
```

Comment:
i was thinking something a bit more like
`Example: segment-8806931859563747931_1160_000_1180_000.tfrecord`
that's the standard file name format, right?

Reply:
Yeah that's the file format, but `record_dir` is the path to the directory that has all these files.